workflow/controller/controller.go

```diff
@@ -189,6 +193,33 @@ func (wfc *WorkflowController) podLabeler(stopCh <-chan struct{}) {
 	}
 }
 
+// podGarbageCollector will delete all pods on the controllers gcPods channel as completed
+func (wfc *WorkflowController) podGarbageCollector(stopCh <-chan struct{}) {
+	for {
+		select {
+		case <-stopCh:
+			return
+		case pod := <-wfc.gcPods:
+			parts := strings.Split(pod, "/")
+			if len(parts) != 2 {
+				log.Warnf("Unexpected item on gcPods channel: %s", pod)
+				continue
+			}
+			namespace := parts[0]
+			podName := parts[1]
+			err := common.DeletePod(wfc.kubeclientset, podName, namespace)
+			if err != nil {
+				// because of multiple delete situation, we can ignore NotFound error safely
+				if !apierr.IsNotFound(err) {
```

Comment:
apierr.IsNotFound(err) should be moved inside `DeletePod` so this check can be removed.